modules/issue_tracker/ajax/EditIssue.php

```diff
@@ -0,0 +1,619 @@
+    if ($validatedInput['isValidSubmission']) {
+        if (array_key_exists('sessionID', $validatedInput)) {
+            $issueValues['sessionID'] = $validatedInput['sessionID'];
+        }
+        if (array_key_exists('candID', $validatedInput)) {
+            $issueValues['candID'] = $validatedInput['candID'];
+        }
+    } else {
+        return $validatedInput;
+    } //aka when it's not valid
+
+    if (!empty($issueID)) {
+        $db->update('issues', $issueValues, ['issueID' => $issueID]);
+    } else {
+        $issueValues['reporter'] = $user->getData('UserID');
+        $issueValues['dateCreated'] = date('Y-m-d H:i:s');
+        $db->insert('issues', $issueValues);
+        $issueID = $db->getLastInsertId();
+    }
+
+    updateHistory($issueValues, $issueID);
+
+    //adding comment in now that I have an issueID for both new and old.
+    if ($_POST['comment'] != null) {
+        $commentValues = array(
+            'issueComment' => $_POST['comment'],
+            'addedBy' => $user->getData('UserID'),
+            'issueID' => $issueID,
+        );
+        $db->insert('issues_comments', $commentValues);
+    }
+
+    if (isset($_POST['othersWatching'])) {
+
+    }
+    //adding editor to the watching table unless they don't want to be added.
+    if ($_POST['watching'] != 'no') {
+        $nowWatching = array(
+            'userID' => $user->getData('UserID'),
+            'issueID' => $issueID,
+        );
+        $db->replace('issues_watching', $nowWatching);
+    } else if ($_POST['watching'] == 'no') {
+        $db->delete(
+            'issues_watching',
+            array(
+                'issueID' => $issueID,
+                'userID' => $user->getData('UserID'),
+            )
+        );
+    }
+
+    //adding others from multiselect to watching table.
+    if (isset($_POST['othersWatching'])) {
+        $othersNowWatching = explode(',', $_POST['othersWatching']);
+        foreach ($othersNowWatching as $userWatching) {
+            if ($userWatching) { //cause sometimes it sends null
+                $nowWatching = array(
+                    'userID' => $userWatching,
+                    'issueID' => $issueID,
+                );
+                $db->replace('issues_watching', $nowWatching);
+            }
+        }
+    }
+
+    //sending email
+    emailUser($issueID);
+
+    return array(
+        'isValidSubmission' => true,
+        'issueID' => $issueID,
+    );
+}
+
+//TODO: change to reflect the fact that
+//TODO: if you have a visitlabel input then you don't need to insert a session
+//also make this better
+/**
+ * Validates those fields that need it
+ * Currently PSCID and visitLabel as fk to session
+ *
+ * @param array $validateValues values to be validated
+ * @param int $issueID issue ID
+ *
+ * @throws DatabaseException
+ *
+ * @return array success with values, or error with message.
+ */
+function validateInput($validateValues, $issueID)
```

Comment:
@alisterdev No, because they want functionality to submit a PSCID without a visit label, and then to be able to change either of those without changing the other. But yes, it is a little ridiculous :P